src/DotNetWorker.Core/Context/FunctionContextHttpRequestExtensions.cs

```diff
@@ -21,16 +22,9 @@ public static class FunctionContextHttpRequestExtensions
         /// <returns>HttpRequestData instance if the invocation is http, else null</returns>
         public static async ValueTask<HttpRequestData?> GetHttpRequestDataAsync(this FunctionContext context)
         {
-            var httpTriggerBinding = context.FunctionDefinition.InputBindings.Values
-                                            .FirstOrDefault(a => a.Type == "httpTrigger");
+            IHttpRequestDataFeature? httpRequestDataFeature = context.Features.Get<IHttpRequestDataFeature>() ?? DefaultHttpRequestDataFeature.Instance;
```

Comment:
Q: Should we move the part where we use the singleton instance to the place where we setup the invocation pipeline ? I see that we do similar things in the aspnet core integration path (doing that in the middleare). This will allow folks to use the feature If they want to use it (Can be more unit test friendly than the extension method)

Reply:
This was discussed with @fabiocav early in the design stage, but there are some changes/refactoring needed that are a bit out of scope to set up this (and other features) by default in the invocation pipeline.